ngraph/test/type_prop/einsum.cpp

```diff
@@ -0,0 +1,291 @@
+
+TEST(type_prop, einsum_staticshape_multimatmul)
+{
+    std::string equation = "ab,bcd,bc->ca";
+    Shape input1_shape{2, 5};
+    Shape input2_shape{5, 3, 6};
+    Shape input3_shape{5, 3};
+    Shape out_shape{3, 2};
+    auto I1 = make_shared<op::Parameter>(element::f32, input1_shape);
+    auto I2 = make_shared<op::Parameter>(element::f32, input2_shape);
+    auto I3 = make_shared<op::Parameter>(element::f32, input3_shape);
+    auto O = make_shared<op::v7::Einsum>(OutputVector{I1, I2, I3}, equation);
+    ASSERT_EQ(O->get_element_type(), element::f32);
+    ASSERT_EQ(O->get_shape(), out_shape);
+}
+
+TEST(type_prop, einsum_staticshape_ellipsis)
+{
+    std::string equation = "a...->...";
+    Shape input1_shape{3, 3};
+    Shape out_shape{3};
+    auto I1 = make_shared<op::Parameter>(element::f32, input1_shape);
+    auto O = make_shared<op::v7::Einsum>(OutputVector{I1}, equation);
+    ASSERT_EQ(O->get_element_type(), element::f32);
+    ASSERT_EQ(O->get_shape(), out_shape);
+}
+
+TEST(type_prop, einsum_staticshape_ellipsis2)
+{
+    std::string equation = "a...,...->a...";
+    Shape input1_shape{3, 3};
+    Shape input2_shape{1};
+    Shape out_shape{3, 3};
+    auto I1 = make_shared<op::Parameter>(element::f32, input1_shape);
+    auto I2 = make_shared<op::Parameter>(element::f32, input2_shape);
+    auto O = make_shared<op::v7::Einsum>(OutputVector{I1, I2}, equation);
+    ASSERT_EQ(O->get_element_type(), element::f32);
+    ASSERT_EQ(O->get_shape(), out_shape);
+}
+
+TEST(type_prop, einsum_staticshape_ellipsis3)
+{
+    std::string equation = "a...b,b...->a...";
+    Shape input1_shape{11, 1, 4, 3};
+    Shape input2_shape{3, 11, 7, 1};
+    Shape out_shape{11, 11, 7, 4};
+    auto I1 = make_shared<op::Parameter>(element::f32, input1_shape);
+    auto I2 = make_shared<op::Parameter>(element::f32, input2_shape);
+    auto O = make_shared<op::v7::Einsum>(OutputVector{I1, I2}, equation);
+    ASSERT_EQ(O->get_element_type(), element::f32);
+    ASSERT_EQ(O->get_shape(), out_shape);
+}
+
+TEST(type_prop, einsum_dynamicshape_dotproduct)
+{
+    std::string equation = "a,ab->ab";
+    const auto input1_shape = PartialShape{Dimension(2, 7)};
+    const auto input2_shape = PartialShape{Dimension(3, 10), 3};
+    const auto out_shape = PartialShape{Dimension(3, 7), 3};
+    auto I1 = make_shared<op::Parameter>(element::f32, input1_shape);
+    auto I2 = make_shared<op::Parameter>(element::f32, input2_shape);
+    auto O = make_shared<op::v7::Einsum>(OutputVector{I1, I2}, equation);
+    ASSERT_EQ(O->get_element_type(), element::f32);
+    ASSERT_TRUE(O->get_output_partial_shape(0).same_scheme(out_shape));
+}
+
+TEST(type_prop, einsum_dynamicshape_diagextraction)
+{
+    std::string equation = "xyzxy->xyz";
+    const auto input1_shape = PartialShape{Dimension(2, 7), Dimension(1, 5), 4, Dimension(3, 5), 3};
+    const auto out_shape = PartialShape{Dimension(3, 5), 3, 4};
+    auto I1 = make_shared<op::Parameter>(element::f32, input1_shape);
```

Comment:
Change the input tensor type to `i32` for some tests

Reply:
Fixed.